libraries/tock-register-interface/README.md

```diff
@@ -190,79 +167,76 @@ register_bitfields! [
-.read(field: Field<T, R>) -> T                 // Read the value of the given field
-.read_as_enum<E>(field: Field<T, R>) -> Option<E> // Read value of the given field as a enum member
-.write(value: FieldValue<T, R>)                // Write the value of one or more fields,
-                                               //  overwriting other fields to zero
-.modify(value: FieldValue<T, R>)               // Write the value of one or more fields,
-                                               //  leaving other fields unchanged
-.modify_no_read(                               // Write the value of one or more fields,
-      original: LocalRegisterCopy<T, R>,       //  leaving other fields unchanged, but pass in
-      value: FieldValue<T, R>)                 //  the original value, instead of doing a register read
-.is_set(field: Field<T, R>) -> bool            // Check if one or more bits in a field are set
-.any_matching_bits_set(value: FieldValue<T, R>) -> bool  // Check if any bits corresponding to the mask in the passed field are set
-.matches_all(value: FieldValue<T, R>) -> bool  // Check if all specified parts of a field match
-.matches_any(&self, fields: &[FieldValue<T, R>]) -> bool // Check if any specified parts of a field match
-.extract() -> LocalRegisterCopy<T, R>          // Make local copy of register
-
-Aliased<T: UIntLike, R: RegisterLongName = (), W: RegisterLongName = ()>: Readable + Writeable
-.get() -> T                                    // Get the raw register value
-.set(value: T)                                 // Set the raw register value
-.read(field: Field<T, R>) -> T                 // Read the value of the given field
-.read_as_enum<E>(field: Field<T, R>) -> Option<E> // Read value of the given field as a enum member
-.write(value: FieldValue<T, W>)                // Write the value of one or more fields,
-                                               //  overwriting other fields to zero
-.is_set(field: Field<T, R>) -> bool            // Check if one or more bits in a field are set
-.any_matching_bits_set(value: FieldValue<T, R>) -> bool  // Check if any bits corresponding to the mask in the passed field are set
-.matches_all(value: FieldValue<T, R>) -> bool  // Check if all specified parts of a field match
-.matches_any(&self, fields: &[FieldValue<T, R>]) -> bool // Check if any specified parts of a field match
-.extract() -> LocalRegisterCopy<T, R>          // Make local copy of register
+tock_registers::peripheral! {
+    Registers {
+        0x0 => a: u8 { Read, Write },
+    }
+}
 ```
 
-The `Aliased` type represents cases where read-only and write-only registers,
-with different meanings, are aliased to the same memory location.
+`peripheral!` generates:
+
+```rust
+trait Registers {
+    type a<'s>: tock_registers::Register<DataType = u8> +
+            tock_registers::Read<LongName = ()> +
+            tock_registers::Write<LongName = ()> where Self: 's;
+    fn a(&self) -> Self::a<'_> { ... }
+}
+
```

Comment:
Can you show an example with the bitfield parameter too?

Reply:
There is an example later, under the "Example: Using registers and bitfields" heading. Maybe I should move the "Register Trait Bounds" heading to the end, so "Example: Using registers and bitfields" would come right after "Defining bitfields"?

Reply:
I don't see a macro expansion for the bitfield parameter under that heading.

Reply:
Oh, I misunderstood Branden's request. I added an register with a bitfield parameter and its trait expansion to the "Register Trait Bounds" section.